src/Layer/Layer.js

```diff
@@ -98,15 +100,16 @@ class Layer extends THREE.EventDispatcher {
             throw new Error(`Layer ${id} needs Source`);
         }
         super();
+        this.isLayer = true;
+
         if (config.style && !(config.style instanceof Style)) {
             if (typeof config.style.fill?.pattern === 'string') {
                 console.warn('Using style.fill.pattern = { source: Img|url } is adviced');
                 config.style.fill.pattern = { source: config.style.fill.pattern };
             }
             config.style = new Style(config.style);
         }
-        this.isLayer = true;
-
+        this.style = config.style || userStyle;
```

Comment:
I think we should avoid using the same instance here. There might indeed be a bug if a user does not define a style for two layers (that will therefore share the same Style instance) and if one then decides to change the style of one of the layer, this style will be applied to both layers.